src/objects.c

```diff
@@ -1069,6 +1069,9 @@ static void dns_connect(struct PgSocket *server)
 		int count = 1;
 		int n;
 
+		if (!server->pool->last_connect_failed && server->pool->db->host_strategy == LAST_SUCCESSFUL)
+			server->pool->rrcounter = server->pool->last_successful_rrcounter;
```

Comment:
The current PR seems to have the desired effect based on the tests you wrote, but I have the  feeling like the same could probably be achieved without the two new fields you added: `server->rrcounter` and `pool->last_successful_rrcounter`.

How about we simply increase the `pool->rr_counter` whenever a connection attempt fails.

Reply:
Thanks for the feedback. I simplified the logic as you suggest and it looks better to me.